codegen/templates/http_client.tmpl

```diff
@@ -124,6 +133,20 @@ func {{$exportName}}(deps *module.Dependencies) Client {
 	}
 }
 
+{{if $sidecarRouter -}}
+func initializeAltRoutingMap(altServiceDetail config.AlternateServiceDetail) map[string]map[string]string {
+	routingMap := make(map[string]map[string]string)
+	for _, alt := range altServiceDetail.RoutingConfigs {
+		if headerValueToServiceMap, ok := routingMap[alt.HeaderName]; ok {
+			headerValueToServiceMap[alt.HeaderValue] = alt.ServiceName
```

Comment:
we're trying to handle duplicates here with a merge?

Reply:
I was trying to handle one header, can have different end routing destinations depending on the value

For example 

		"routingConfigs": [
			{
				"headerName": "Rpc-Header-Dest",
				"headerValue": "some-value1",
				"serviceName": "some-service"
			},
			{
				"headerName": "Rpc-Header-Dest",
				"headerValue": "another-value2",
				"serviceName": "another-service"
			}

You can argue that we don't need to support this, but its what I was thinking.